app/main.py

```diff
@@ -2,5 +2,26 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_dict = {}
+
+    def wrapper(*args, **kwargs) -> None:
+        name_func = func.__name__
+        if name_func in result_dict:
+            if args in [arg[0] for arg in result_dict[name_func]]:
+                print("Getting from cache")
+                result_num = [num for num in result_dict[name_func]
```

Comment:
why do you need it?

Reply:
Save the results in a dictionary  **result_dict** {'name_func':[(arg),result_num]}.
line 9 - checking the key in the dictionary
line 10 - checking in a dictionary argument with that function name
line 12 - take the result_num of the execution of the function without starting the re-execution of the function

Reply:
Sorry, the dictionary will not be for all functions, but for each. name_func no need to save.